tools/mo/unit_tests/mo/convert/version_checker_test_actual.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import unittest
+
+from openvino.runtime import get_version as get_ie_version
+
+from openvino.tools.mo.utils.version import get_version, get_simplified_ie_version, \
+    get_simplified_mo_version, VersionChecker
+
+
+class VersionCheckerTest(unittest.TestCase):
+    def test_version_checker(self):
+        import datetime
+        import os
+        ref_mo_version = get_version()
+        ref_ie_version = get_ie_version()
+        ref_mo_simplified_version = get_simplified_mo_version()
+        ref_ie_simplified_version = get_simplified_ie_version(env=os.environ)
+
+        # first init of VersionChecker
+        start_time = datetime.datetime.now()
+        VersionChecker().check_runtime_dependencies()
+        VersionChecker().get_mo_version()
+        VersionChecker().get_ie_version()
+        VersionChecker().get_mo_simplified_version()
+        VersionChecker().get_ie_simplified_version()
+        first_init_time = (datetime.datetime.now() - start_time).total_seconds()
+
+        # Loop with multiple usages of VersionChecker
+        start_time = datetime.datetime.now()
+        for _ in range(100):
+            VersionChecker().check_runtime_dependencies()
+            assert VersionChecker().get_mo_version() == ref_mo_version
+            assert VersionChecker().get_ie_version() == ref_ie_version
+            assert VersionChecker().get_mo_simplified_version() == ref_mo_simplified_version
+            assert VersionChecker().get_ie_simplified_version() == ref_ie_simplified_version
+        loop_time = (datetime.datetime.now() - start_time).total_seconds()
+
+        # Check that time of loop is less than first init, so no actual initialization happens
+        assert loop_time < first_init_time
```

Comment:
I would say that loop_time / 100 is less than `first_init_time`. Please check

Reply:
Both works, as first init time is significantly longer than even whole loop. On my machine I get following time:

first init time: 0.26 sec
loop time: 7.4e-05 sec
loop time / 100: 7.4e-07 sec

So I can change the check to `loop_time / 100`, but initial check looks safe to use too.

Reply:
I am saying that just to avoid sporadic failures in CI, for example